ffi/newpassmanagers.cpp

```diff
@@ -357,4 +514,70 @@ LLVMPY_buildFunctionSimplificationPipeline(LLVMPassBuilderRef PBref,
     return llvm::wrap(FPM);
 }
 
+API_EXPORT(void)
+LLVMPY_module_AddModuleDebugInfoPrinterPass(LLVMModulePassManagerRef MPM) {
+    llvm::unwrap(MPM)->addPass(ModuleDebugInfoPrinterPass(llvm::outs()));
+}
+
+API_EXPORT(void)
+LLVMPY_module_AddVerifierPass(LLVMModulePassManagerRef MPM) {
+    llvm::unwrap(MPM)->addPass(VerifierPass());
+}
+
+#define CGSCC_PASS(NAME)                                                       \
+    void LLVMPY_module_Add##NAME(LLVMModulePassManagerRef MPM) {               \
+        llvm::unwrap(MPM)->addPass(                                            \
+            createModuleToPostOrderCGSCCPassAdaptor(NAME()));                  \
+    }
+#include "PASSREGISTRY.def"
+
+#define MODULE_PASS(NAME)                                                      \
+    void LLVMPY_module_Add##NAME(LLVMModulePassManagerRef MPM) {               \
+        llvm::unwrap(MPM)->addPass(NAME());                                    \
+    }
+#include "PASSREGISTRY.def"
+
+#define FUNCTION_PASS(NAME)                                                    \
+    void LLVMPY_module_Add##NAME(LLVMModulePassManagerRef MPM) {               \
+        llvm::unwrap(MPM)->addPass(createModuleToFunctionPassAdaptor(NAME())); \
+    }                                                                          \
+    void LLVMPY_function_Add##NAME(LLVMFunctionPassManagerRef FPM) {           \
+        llvm::unwrap(FPM)->addPass(NAME());                                    \
+    }
+#include "PASSREGISTRY.def"
+
+#define LOOP_PASS(NAME)                                                        \
+    void LLVMPY_module_Add##NAME(LLVMModulePassManagerRef MPM) {               \
+        llvm::unwrap(MPM)->addPass(createModuleToFunctionPassAdaptor(          \
+            createFunctionToLoopPassAdaptor(NAME())));                         \
+    }                                                                          \
+    void LLVMPY_function_Add##NAME(LLVMFunctionPassManagerRef FPM) {           \
+        llvm::unwrap(FPM)->addPass(createFunctionToLoopPassAdaptor(NAME()));   \
+    }
+#include "PASSREGISTRY.def"
+
+const char *LLVMPY_getModuleLevelPasses() {
```

Comment:
This probably needs an `API_EXPORT` as well but I don't trust myself to get the syntax exactly right without trying it first, so no suggestion as such :-)